pkg/lib/authn/otp/form.go

```diff
@@ -35,30 +35,50 @@ func (f Form) GenerateCode(cfg *config.TestModeConfig, featureCfg *config.TestMo
 	codeType := f.codeType()
 	switch c := codeType.(type) {
 	case secretcode.OOBOTPSecretCodeType:
-		if cfg.FixedOOBOTP.Enabled {
-			if r, ok := cfg.FixedOOBOTP.MatchTarget(target); ok {
-				fixedOTP := r.FixedCode
-				if fixedOTP == "" {
-					fixedOTP = featureCfg.FixedOOBOTP.Code
-				}
-				if fixedOTP == "" {
-					fixedOTP = c.Generate()
-				}
-				return c.GenerateFixed(fixedOTP)
+		return f.generateOOBOTPCode(c, cfg, featureCfg, target)
+	case secretcode.LinkOTPSecretCodeType:
+		return f.generateLinkOTPCode(c, cfg, featureCfg, target, userID)
+	default:
+		panic("unknown otp form")
+	}
+}
```

Comment:
Please add tests for GenerateCode to ensure we do not have regression.

By looking at the code, I see an interesting behavior.

1. Test mode is enabled
2. A rule is matched.
3. The rule does not specify fixed code
4. Feature config test mode is disabled.
5. But Feature config test mode has fixed code defined.
6. The fixed code is taken from the feature config (but it is disabled).